### PR TITLE
test_models: fuzz test panda and CarState

### PR DESCRIPTION
**Pull request by sshane**
Actually finding lots of legitimate mismatches between openpilot/CANParser and panda safety:

- CANParser issues:
  - [x] Honda: `CANParser` maintains values from invalid msgs in `vl_all` when it shouldn't add them in the first place, causing bad brake press values in CS when it gets a valid msg - https://github.com/commaai/opendbc/pull/977
  - [x] VW MQB DBC has `CHECKSUM` before `COUNTER`, leading to counter not always getting updated properly if checksum is invalid - https://github.com/commaai/opendbc/pull/977
  - [x] CANParser's invalid counter counter wraps around since it's not clipped. It also takes a while to go valid after becoming invalid since it's not clipped - https://github.com/commaai/opendbc/pull/976
- panda issues:
  - [x] ~VW PQ: standstill checks don't match, panda doesn't check counter for `MSG_BREMSE_1` but `CANParser` does~ **dashcam!**
  - [x] Ford CS checks standstill signal == 1, safety checks != 0, but signal is 2 bits - https://github.com/commaai/panda/pull/1725
  - [x] panda safety doesn't check interceptor counter (Honda & Toyota) - ~https://github.com/commaai/panda/pull/1738~ https://github.com/commaai/panda/pull/1735
  - [x] Hyundai brake pressed is 2 bits, safety bitmasks 0x2 instead of 0x3. Value 3 is never seen in the fleet date, so never would be picked up with real routes. - https://github.com/commaai/panda/pull/1724
  - [x] ~Tesla CS and safety check two different messages for `standstill`/`vehicle_moving`, probably because `ESP_B` isn't on both PT and the other bus. @robbederks can we just make openpilot use `DI_torque2`? This also fixes possible mismatches for the angle safety since it uses vehicle speed - https://github.com/commaai/opendbc/issues/1772~ **dashcam!**
  - [x] Honda Bosch alt brake: added `BRAKE_MODULE` check to the same set as `POWERTRAIN_DATA` which always exists, so it randomly selects an address and only checks one message (https://github.com/commaai/panda/pull/649) - https://github.com/commaai/panda/pull/1746
  - [x] dynamic rx check fields aren't reset on safety mode init - https://github.com/commaai/panda/pull/1767
  - [x] Honda: prev brake switch value isn't reset on safety mode init - https://github.com/commaai/panda/pull/1781

### CODE REVIEW

**Comment by sshane** [comment]
Memory leak seems to have resolved with my clean up. Running with thousands of examples and finding a few more issues (a Nissan Rogue), running this for a while before merging.

**Comment by sshane** [comment]
Wanted to understand the apparent memory leak I was seeing running with a lot of examples before merging, turns out it is pytest capturing and storing stdout. Many come from here in opendbc: https://github.com/commaai/opendbc/blob/93b983d49a2d6d5c67e15ce7650f55e4f121485d/can/parser.cc#L85-L87

Disabling capturing solves the memory leak and you can run with thousands of examples now!

**Comment by sshane** [comment]
Last 5 runs:

| branch | jenkins | cars (0) | cars (1) | cars (2) | cars (3) | cars (4) |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| `master`  | 1:45 | 4:17 | 4:12 | 3:57 | 3:45 | 4:13 |
| `fuzzy-panda-carstate`  | 3:17 (+1:32) | 5:21 (+1:04) | 5:22 (+1:10) | 4:55 (+0:58) | 4:38 (+0:53) | 5:17 (+1:04) |

# panda-fuzzy
GH Actions:
cars (0): 4:50, 5:13, 5:12, 6:31, 5:01
cars (1): 5:02, 6:23, 5:22, 5:04, 5:03
cars (2): 4:43, 4:39, 4:34, 4:41, 6:01
cars (3): 5:18, 4:34, 4:34, 4:28, 4:18
cars (4): 6:09, 5:03, 5:10, 5:06, 5:00

GH Actions:
cars (0): 4:16, 4:17, 4:16, 4:17, 4:21: 4:17 avg
cars (1): 4:13, 4:14, 4:12, 4:12, 4:10: 4:12 avg
cars (2): 3:57, 3:56, 3:57, 3:57, 3:59: 3:57 avg
cars (3): 3:45, 3:45, 3:46, 3:48, 3:44: 3:45 avg
cars (4): 4:13, 4:16, 4:13, 4:17, 4:09: 4:13 avg

File: selfdrive/car/tests/test_models.py [hunk 1]
```diff
@@ -33,6 +35,7 @@
 JOB_ID = int(os.environ.get("JOB_ID", "0"))
 INTERNAL_SEG_LIST = os.environ.get("INTERNAL_SEG_LIST", "")
 INTERNAL_SEG_CNT = int(os.environ.get("INTERNAL_SEG_CNT", "0"))
+MAX_EXAMPLES = int(os.environ.get('MAX_EXAMPLES', '50'))
```
Comment (sshane) [inline selfdrive/car/tests/test_models.py:38]:
```suggestion
MAX_EXAMPLES = int(os.environ.get("MAX_EXAMPLES", "50"))
```

**Comment by adeebshihadeh** [comment]
This is a significant slowdown to GH actions. Can we lower it to 15s?

**Comment by sshane** [comment]
I measured master and this branch at different times, but the performance you get on GH Actions seems inconsistent through time. Nowon master after this change, it's literally only ~15s slower.

**Comment by adeebshihadeh** [comment]
Ok let's monitor it and reduce if we see slowdowns >15s. I would like all CI to finish <5m soon. We're close to that, but we can't regress too much.

**Pull request merged**